javers-core/src/test/groovy/org/javers/core/ShallowReferenceTest.groovy

```diff
@@ -0,0 +1,65 @@
+package org.javers.core
+
+import org.javers.core.model.Category
+import org.javers.core.model.DummyUser
+import org.javers.core.model.DummyUserContact
+import org.javers.core.model.DummyUserPhone
+import spock.lang.Specification
+
+import static org.javers.core.JaversBuilder.javers
+
+/**
+ * @author akrystian
+ */
+class ShallowReferenceTest extends Specification {
+
+    def "should support ShallowReference"() {
+        given:
+        def left = new DummyUserContact("Tony")
+        def right = new DummyUserContact("Tony")
+        left.setPropertyWithShallowReferenceAnn(bossPhone)
```

Comment:
może przekazać tą referncję przez konstruktor? myślę, że nie wpłynie to znacznie na obniżenie czytelności, wszystko świetnie widać w when